src/Domain/CoordinateMaps/TimeDependent/ShapeMapTransitionFunctions/SphereTransition.cpp

```diff
@@ -37,68 +48,197 @@ SphereTransition::SphereTransition(const double r_min, const double r_max,
 }
 
 double SphereTransition::operator()(
-    const std::array<double, 3>& source_coords) const {
-  return call_impl<double>(source_coords);
+    const std::array<double, 3>& source_coords,
+    const std::optional<size_t>& one_over_radius_power) const {
+  const double mag = magnitude(source_coords);
+
+  if (UNLIKELY(one_over_radius_power.value_or(0_st) >= 3 and
+               equal_within_roundoff(mag, 0.0))) {
+    ERROR("Trying to divide by a point "
+          << source_coords
+          << " with radius zero in SphereTransition operator.");
+  }
+
+  if (interior_) {
+    return inverse_cube_r_min_ *
+           (one_over_radius_power.value_or(0_st) < 3
+                ? integer_pow(mag,
+                              static_cast<int>(
+                                  2 - one_over_radius_power.value_or(0_st)))
+                : 1.0 /
+                      integer_pow(mag, static_cast<int>(
+                                           one_over_radius_power.value() - 2)));
+  }
+
+  if (UNLIKELY(mag < (1.0 - eps_) * r_min_)) {
+    ERROR("SphereTransition coord " << source_coords << " with radius " << mag
+                                    << " is within r_min of " << r_min_
+                                    << ", but the class was not constructed "
+                                       "for the interior of the sphere.");
+  } else if (UNLIKELY(mag > (1.0 + eps_) * r_max_)) {
+    ERROR("SphereTransition coord " << source_coords << " with radius " << mag
+                                    << "is beyond r_max of " << r_max_ << ".");
+  }
+
+  double result = (a_ * mag + b_);
+  // Avoid roundoff
+  result = std::clamp(result, 0.0, 1.0);
+
+  return result /
+         integer_pow(
+             mag, static_cast<int>(1 + one_over_radius_power.value_or(0_st)));
 }
+
 DataVector SphereTransition::operator()(
-    const std::array<DataVector, 3>& source_coords) const {
-  return call_impl<DataVector>(source_coords);
+    const std::array<DataVector, 3>& source_coords,
+    const std::optional<size_t>& one_over_radius_power) const {
+  DataVector result = source_coords[0];
+  // Go point by point
+  for (size_t i = 0; i < source_coords[0].size(); i++) {
+    result[i] = (*this)(std::array{source_coords[0][i], source_coords[1][i],
+                                   source_coords[2][i]},
+                        one_over_radius_power);
+  }
+  return result;
 }
 
 std::optional<double> SphereTransition::original_radius_over_radius(
     const std::array<double, 3>& target_coords,
     double radial_distortion) const {
   const double mag = magnitude(target_coords);
+  // If we are at the center, the radius is the same
+  if (UNLIKELY(equal_within_roundoff(mag, 0.0))) {
+    return interior_ ? std::optional{1.0} : std::nullopt;
+  }
+
+  // a_ being positive is a sentinel for reversed.
+  // If we aren't reversed, check near or within r_min_.
+  if (a_ < 0.0) {
+    if (mag + radial_distortion < (1.0 - eps_) * r_min_) {
```

Comment:
Why the `+ radial_distortion` here?

I think you have to be careful that you compare r_min and r_max only to the original radius, not the target radius. E.g. in line 157 there's a comparison to the target radius, which I think could be a problem.

Reply:
It's undoing the distortion assuming we are at the f=1 surface. Therefore if this condition is true, we are guaranteed to be within the excision and if not, we're outside of it. I ported this check over from SpEC.

Below on 157 should still be the target radius, but I forgot to add the radial distortion since I'm trying to see if we are exactly at the excision (because then the formula is simpler). In the Wedge it is correct